### PR TITLE
[BugFix]Support redundant experts in EPLB

### PR DESCRIPTION
**Pull request by yechao237**
This PR adds support for redundant experts in the EPLB. 

Key points: 
- Use global_num_experts = num_experts + num_redundant_experts consistently. 
- Backward compatible when num_redundant_experts=0. 

Tested 
On a 16-rank setup (W8A8) with static EPLB and expert_map_path, verifying router logits shape and successful requests.

- vLLM version: v0.11.0rc3
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.11.0

### CODE REVIEW

File: vllm_ascend/torchair/quantization/torchair_w8a8_dynamic.py [hunk 1]
```diff
@@ -928,10 +925,8 @@ def apply(
         dynamic_scale_for_share: Optional[Any] = None,
         **kwargs,
     ) -> torch.Tensor:
-        assert router_logits.shape[
-            1] == global_num_experts, "Number of global experts mismatch"
 
-        is_deepseek_v3_r1 = global_num_experts == 256
+        is_deepseek_v3_r1 = global_num_experts - global_redundant_expert_num == 256
```
Comment (offline893) [inline vllm_ascend/torchair/quantization/torchair_w8a8_dynamic.py]: Redundant experts need to be handled as function parameters, and the related code needs to be modified accordingly.
